internal/comparetest/golden/golden.go

```diff
@@ -96,3 +97,33 @@ func WriteLogs(filePath string, logs plog.Logs) error {
 	b = append(b, []byte("\n")...)
 	return os.WriteFile(filePath, b, 0600)
 }
+
+// ReadTraces reads a ptrace.Traces from the specified file
+func ReadTraces(filePath string) (ptrace.Traces, error) {
+	b, err := os.ReadFile(filepath.Clean(filePath))
+	if err != nil {
+		return ptrace.Traces{}, err
+	}
+
+	unmarshaler := ptrace.JSONUnmarshaler{}
+	return unmarshaler.UnmarshalTraces(b)
+}
+
+// WriteTraces writes a ptrace.Traces to the specified file
+func WriteTraces(filePath string, traces ptrace.Traces) error {
```

Comment:
Please see updated implementation of `WriteMetrics`. We should force users to pass in `*testing.T` so we can fail tests that use this function.